docker/build-tools/Dockerfile

```diff
@@ -539,11 +596,15 @@ WORKDIR /
 
 ENTRYPOINT ["/usr/local/bin/docker-entrypoint"]
 
+
 ##############
 # Clang+LLVM
 ##############
 
-FROM ubuntu:xenial as clang_context
+FROM ubuntu:xenial AS clang_context_amd64
+FROM ubuntu:focal AS clang_context_arm64
+# hadolint ignore=DL3006
+FROM clang_context_${TARGETARCH} AS clang_context
```

Comment:
Interesting... FROM can evaluate var
TIL

Reply:
I learned this tip from istio repo 
https://github.com/istio/istio/blob/master/operator/docker/Dockerfile.operator#L15